examples/read_nodes.rs

```diff
@@ -0,0 +1,115 @@
+// example that separates logic for reading different top-level nodes of xml tree
+
+use quick_xml::events::{BytesStart, Event};
+use quick_xml::reader::Reader;
+use std::collections::HashMap;
+use std::str;
+
+// Code doesn't work if DefaultSettings node is constructed like this:
+//   <DefaultSettings Language="es" Greeting="HELLO"/>
```

Comment:
This is because 
```xml
<DefaultSettings Language="es" Greeting="HELLO"/>
```
would produce `Empty` event. You could set [`expand_empty_elements`](https://docs.rs/quick-xml/latest/quick_xml/reader/struct.Reader.html#method.expand_empty_elements) to `true` to always get `Start/End` pair and do not worry about it.